src/motion/AccelerometerThread.h

```diff
@@ -0,0 +1,141 @@
+#pragma once
+#ifndef _ACCELEROMETER_H_
+#define _ACCELEROMETER_H_
+
+#include "configuration.h"
+
+#if !defined(ARCH_PORTDUINO) && !defined(ARCH_STM32WL) && !MESHTASTIC_EXCLUDE_ENVIRONMENTAL_SENSOR
+
+#include "../concurrency/OSThread.h"
+#include "BMA423Sensor.h"
+#include "BMX160Sensor.h"
+#include "ICM20948Sensor.h"
+#include "LIS3DHSensor.h"
+#include "LSM6DS3Sensor.h"
+#include "MPU6050Sensor.h"
+#include "MotionSensor.h"
+#include "STK8XXXSensor.h"
+
+class AccelerometerThread : public concurrency::OSThread
+{
+  private:
+    MotionSensor *sensor = nullptr;
+    ScanI2C::FoundDevice device = ScanI2C::DEVICE_NONE;
+    bool isInitialised = false;
+
+  public:
+    explicit AccelerometerThread(ScanI2C::FoundDevice found) : OSThread("AccelerometerThread")
+    {
+        if (found.address.port == ScanI2C::I2CPort::NO_I2C) {
+            LOG_DEBUG("AccelerometerThread disabling due to no sensors found\n");
+            disable();
+            return;
+        }
+
+#ifndef RAK_4631
```

Comment:
Q) Does RAK have magic accelerometers that are always in an interesting configuration? :)

Reply:
4631 always uses tap and double tap functionaltiy when the module is detected